Jetpack App: Add 'jetpack/app-branding' flag

SiobhyB opened the pull request:
Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack. Making these changes behind a feature flag for now will enable us to make progress and ship all necessary changes to Calypso when ready. 

#### Proposed Changes

* This PR introduces a `jetpack/app-branding` feature flag, which was added following [the guidance here](https://github.com/Automattic/wp-calypso/tree/trunk/config#feature-flags). The flag is enabled for `development` and `wpcalyspo` as a starting point, but doesn't change any functionality for now.

#### Testing Instructions

* Verify that these changes don't introduce any unexpected errors when browsing Calypso.
* If possible, confirm that adding the `config.isEnabled('jetpack/app-branding');` works to enable/disable features in development.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

SiobhyB committed:
Add 'jetpack/app-branding' flag

This flag will be enabled for development by default, as we begin to change mentions of the WP app to Jetpack.

SiobhyB committed:
Add flag to 'test.json'

SiobhyB committed:
Set flag to true for 'wpcalypso.json'

This will make it easier for developers to test the changes while working on them.

SiobhyB commented:
Noting that the `E2E Tests (mobile) (Web app)` failure isn't specific to this PR, it's failing for all recent PRs. See for more details: p1661153765781449/1660416713.154689-slack-C02DQP0FP

tyxla commented on config/stage.json:
```diff
@@ -46,6 +46,7 @@
 		"inline-help": true,
 		"jetpack/agency-dashboard": true,
 		"jetpack/api-cache": true,
+		"jetpack/app-branding": false,
```
Feel free to omit the flag in environment configurations when it will be disabled. Feature flags are disabled by default, so if we omit it, it works the same way as if it's specifically disabled.

SiobhyB replied:
Thanks for the heads up! I've gone ahead to remove the flag from environments it's not used in https://github.com/Automattic/wp-calypso/pull/66814/commits/010dd41ab73769c62485b799cab600ce61e949f1. 🙇‍♀️

tyxla commented on config/test.json:
```diff
@@ -44,6 +44,7 @@
 		"importers/substack": true,
 		"inline-help": true,
 		"jetpack/agency-dashboard": true,
+		"jetpack/app-branding": true,
```
Do we need it enabled for the testing environment as well?

SiobhyB replied:
No, that was a mistake on my side. Removed as part of https://github.com/Automattic/wp-calypso/pull/66814/commits/010dd41ab73769c62485b799cab600ce61e949f1. Thanks!

SiobhyB committed:
Remove flag in environments where it's not enabled

As flag are 'false' by default, it's okay to omit them in environments where they're not used.

tyxla reviewed (approved):
This looks great 👍 Excited to see the work on the WP.com functionality be moved to the Jetpack app 💯
